stagecoach/price-pusher/price_pusher/configs/price_config.py

```diff
@@ -0,0 +1,134 @@
+from typing import List, Optional
+
+import yaml
+from pragma.core.assets import (
+    PragmaFutureAsset,
+    PragmaAsset,
+    PragmaSpotAsset,
+    AssetType,
+    get_asset_spec_for_pair_id_by_type,
+)
+from pydantic import BaseModel, ConfigDict, Field, field_validator
+from typing_extensions import Annotated
+
+
+class PairConfig(BaseModel):
+    spot: Optional[List[PragmaSpotAsset]] = None
+    future: Optional[List[PragmaFutureAsset]] = None
+
+
+class PriceConfig(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    pairs: PairConfig
+    time_difference: Annotated[int, Field(strict=True, gt=0)]
+    price_deviation: Annotated[float, Field(strict=True, gt=0)]
+
+    @field_validator("pairs", mode="before")
+    def validate_pairs(cls, value: PairConfig) -> PairConfig:
+        if "spot" in value:
+            value["spot"] = cls.validate_asset_pairs(value["spot"], "SPOT")
+        if "future" in value:
+            value["future"] = cls.validate_asset_pairs(value["future"], "FUTURE")
+        return value
+
+    @staticmethod
+    def validate_asset_pairs(pairs: List[str], asset_type: AssetType) -> List:
+        assets = []
+        for pair in pairs:
+            pair = pair.replace(" ", "").upper()
+            splitted = pair.split("/")
+            if len(splitted) != 2:
+                raise ValueError("Pair should be formatted as 'BASE/QUOTE'")
+            asset = get_asset_spec_for_pair_id_by_type(pair, asset_type)
+            assets.append(asset)
+        return assets
+
+    @classmethod
+    def from_yaml(cls, path: str) -> List["PriceConfig"]:
+        with open(path, "r") as file:
+            price_configs = yaml.safe_load(file)
+        list_configs = [cls(**config) for config in price_configs]
+        # TODO: verify that pairs are unique among groups
+        return list_configs
+
+    def get_unique_spot_assets(self) -> List[PragmaSpotAsset]:
```

Comment:
The Assets are not serializable so we can't hash them into a Set

Reply:
can we make them serializable?

Reply:
not really since the underlying type is a dict, would require some tricks (we can't implement __hash__ on TypedDict, no effect)

we could probably do it by using dataclasses instead of TypedDict:
```python
@dataclass
class PragmaSpotAsset:
    type: str
    pair: Tuple[str, str]
    decimals: int

    def __hash__(self):
        return hash((self.type, self.pair, self.decimals))
```

this would work but requires more in depth refactoring